guardian/frontend
# Fetch tests from google docs

## What does this change?
Allows epic copy to be controlled from this doc: https://docs.google.com/spreadsheets/d/1Hoqzg-LeB0xJf2z0JwsfDTHdXKtq-7O5DsQhpqRm7ho/edit#gid=0

Using this tool from the Visuals team: https://visuals.gutools.co.uk/docs/

Our ultimate goal is to remove engineering involvement in putting out epic tests, and our preferred solution is to serve the epic via DFP, which @Mullefa has been working on (see #19668 and guardian/commercial-templates/pull/200). But pending completion of this work, we'd like a stopgap solution that doesn't deviate too far from the existing framework.

So I've created four separate tests, each of which contains one more variant than the last. This is based on the assumption that we will only run one copy test at a time, but we don't know in advance how many variants the test will have. So whoever's running the test can use the switchboard to turn on whichever test they need.

## How to use (for non-engineers)
1. Edit copy in https://docs.google.com/spreadsheets/d/1Hoqzg-LeB0xJf2z0JwsfDTHdXKtq-7O5DsQhpqRm7ho/edit#gid=0

#### test in CODE env:
2. Go to https://frontend.code.dev-gutools.co.uk/dev/switchboard and turn on the correct test
3. Go to https://m.code.dev-theguardian.com/environment/2016/apr/29/revitalised-and-calmed-by-an-english-wood-in-spring#experiments
4. Click left-hand sidebar, choose variant, click reload

| Sidebar button  | A/B tests |
| ------------- | ------------- |
| ![picture 215](https://user-images.githubusercontent.com/5122968/41411000-310373c6-6fd3-11e8-990f-0051b908d634.png) | ![picture 214](https://user-images.githubusercontent.com/5122968/41411001-3119fd80-6fd3-11e8-8792-22dfed4bc9a3.png)  |

5. Check variant looks OK

#### switch on in PROD:
6. Click "Publish" in https://visuals.gutools.co.uk/docs/. It may take a while and a few button clicks, but eventually the "Prod" icon will go green
7. At this point, you can switch on your test here: https://frontend.gutools.co.uk/dev/switchboard. Make sure only **one** of the Google Docs tests is ticked


## Other useful info/caveats/limitations (for non-engineers)
- You can omit the testimonial by leaving **testimonialName** and **testimonialText** blank
- You cannot change the highlight line at the end
- You **must not** change the text of the headings in the spreadsheet (**header**, **p1**, **p2** etc), or the names of the tabs, otherwise you'll just get the control copy in your variant
- While testing in the CODE environment, you don't need to click "Publish" to see changes made in the doc, but they may take a few minutes to appear
- Each test gets variants starting from 1, i.e. `ab-acquisitions-epic-from-google-doc-one-variant` will fetch copy from tab **variant_1**, `ab-acquisitions-epic-from-google-doc-two-variants` will fetch from **variant_1** and **variant_2**, etc

## Screenshots (based on current Google Docs copy - obviously this will change when used for real tests)
### Control
![picture 209](https://user-images.githubusercontent.com/5122968/41409803-593956a2-6fcf-11e8-9de2-395fc03a6f77.png)
### Variant One
![picture 210](https://user-images.githubusercontent.com/5122968/41409802-59224bd8-6fcf-11e8-949d-849510ded06f.png)
### Variant Two
![picture 211](https://user-images.githubusercontent.com/5122968/41409800-590d75b4-6fcf-11e8-855c-4ff4fd91f8a9.png)
### Variant Three
![picture 212](https://user-images.githubusercontent.com/5122968/41409799-58f580ee-6fcf-11e8-8b0c-f39ea574c928.png)
### Variant Four
![picture 213](https://user-images.githubusercontent.com/5122968/41409798-58dae270-6fcf-11e8-8097-5954f7b025e9.png)
